spec/test/collections/test_silva.py

```diff
@@ -0,0 +1,266 @@
+            ), (
+                {
+                    # enum
+                    'id': 'id',
+                    'name': 'name',
+                    'rank': 'superclass',
+                    'dataset': ['nr99', 'ref', 'parc'],  # array in wrong order
+                },
+                "['nr99', 'ref', 'parc'] is not one of [['parc'], ['parc', 'ref'], ['parc', 'ref', 'nr99']]",
+            )
+        ]
+
+        cls.edges_valid = [
+            {
+                'id': '2',
+                'from': '2',
+                'to': '0',
+            }, {
+                'id': '42919',
+                'from': '42919',
+                'to': '2',
+            }, {
+                'id': 'HM032797.1.1344',
+                'from': 'HM032797.1.1344',
+                'to': '44300',
+            }, {
+                'id': 'CRQV01000019.5091.6588',
+                'from': 'CRQV01000019.5091.6588',
+                'to': '1853',
+            },
+        ]
+
+        cls.edges_invalid_errors = [
+            (
+                {
+                    # missing
+                    'from': '2',
+                    'to': '0',
+                },
+                "'id' is a required property",
+            ), (
+                {
+                    # missing
+                    'id': '2',
+                    'to': '0',
+                },
+                "'from' is a required property",
+            ), (
+                {
+                    # missing
+                    'id': '2',
+                    'from': '2',
+                },
+                "'to' is a required property",
+            ), (
+                {
+                    # type
+                    'id': 2,
+                    'from': '2',
+                    'to': '0',
+                },
+                "2 is not of type 'string'",
+            ), (
+                {
+                    # type
+                    'id': '2',
+                    'from': 2,
+                    'to': '0',
+                },
+                "2 is not of type 'string'",
+            ), (
+                {
+                    # type
+                    'id': '2',
+                    'from': '2',
+                    'to': 0,
+                },
+                "0 is not of type 'string'",
+            )
+        ]
+
+    def _test_type(self, schema, valid, invalid_errors):
+        for inst in valid:
+            validate(inst, schema=schema)
+
+        for inst, err_expected in invalid_errors:
+            with self.assertRaises(ValidationError) as cm:
+                validate(inst, schema=schema)
+            msg = str(cm.exception).split('\n')[0]
+
+            print(msg)
+            self.assertTrue(msg == err_expected, '`%s` vs `%s`' % (msg, err_expected))
+
+    def test(self):
+        self._test_type(self.schema_node, self.nodes_valid, self.nodes_invalid_errors)
+        self._test_type(self.schema_edge, self.edges_valid, self.edges_invalid_errors)
```

Comment:
If you're running a lot of similar tests, it's worth using `unittest`'s `subTest()` method; it means that if one of the tests fails, it will continue testing the next input, rather than just dying. See the docs: https://docs.python.org/3.6/library/unittest.html#distinguishing-test-iterations-using-subtests